tools/python/liveanalytics_migration_scripts/unload/utils/s3_utils.py

```diff
@@ -228,3 +233,79 @@ def wait_for_multipart_uploads(
         raise TimeoutError(
             f"Multipart uploads did not complete after {max_attempts} attempts"
         )
+
+    def sync_line_protocol_to_storage(
+        self,
+        s3_bucket_path: str,
+        directory: str,
+        timestream_database_name="",
+        timestream_table_name="",
+    ):
+        """
+        Downloads line protocol data from an S3 bucket path to a directory.
+        If provided simply the bucket path and the Timestream database and
+        table name, the latest unload directory will be searched for. If the
+        S3 bucket path is a path within an S3 bucket, such as s3://my-bucket/my-path,
+        then all objects will be downloaded from this path.
+
+        Args:
+            s3_bucket_path (str): The path of the S3 bucket to download objects from,
+                for example, s3://my-bucket, or, s3://my-bucket/my-path.
+            directory (str): The path to the local directory to download objects to.
+            timestream_database_name (str): The name of the Timestream for LiveAnalytics
+                database used in the unload process.
+            timestream_table_name (str): The name of the Timestream for LiveAnalytics
+                table used in the unload process.
+        Returns:
+            None
+        """
+        if s3_bucket_path.lower().startswith("s3://"):
+            s3_bucket_path = s3_bucket_path[5:]
+        s3_bucket_parts = s3_bucket_path.split("/")
+        if not s3_bucket_parts:
+            raise RuntimeError("S3 bucket path was empty")
+        s3_bucket_name = s3_bucket_parts[0]
+        if not self.s3_bucket_exists(s3_bucket_name):
+            raise RuntimeError(f"S3 bucket {s3_bucket_name} does not exist")
+
+        os.makedirs(directory, exist_ok=True)
+
+        if len(s3_bucket_parts) > 1:
+            prefix_parts = "/".join(s3_bucket_parts[1:])
```

Comment:
`prefix_parts` is already the correct prefix, we can remove the below line and rename this to `prefix`
```suggestion
            prefix = "/".join(s3_bucket_parts[1:])
```